src/leo_object_storage_diskspace_mon.erl

```diff
@@ -0,0 +1,168 @@
+          check_intervals = timer:minutes(1) :: pos_integer(),
+          avs_path_and_servers_list = [] :: [#avs_path_and_servers{}]
+         }).
+
+-define(DEF_TIMEOUT, timer:seconds(30)).
+
+
+%%====================================================================
+%% API
+%%====================================================================
+%% @doc Starts the server with check intervals (sec)
+%%
+-spec(start_link(CheckIntervals, AvsPathAndServersList) ->
+             {ok, pid()} | {error, any()} when CheckIntervals::pos_integer(),
+                                               AvsPathAndServersList::[#avs_path_and_servers{}]).
+start_link(CheckIntervals, AvsPathAndServersList) ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [CheckIntervals, AvsPathAndServersList], []).
+
+
+%% @doc Stop this server
+%%
+-spec(stop() ->
+             ok).
+stop() ->
+    gen_server:call(?MODULE, stop, ?DEF_TIMEOUT).
+
+
+%%====================================================================
+%% GEN_SERVER CALLBACKS
+%%====================================================================
+%% @doc Initiates the server
+init([CheckIntervals, AvsPathAndServersList]) ->
+    erlang:send_after(CheckIntervals, self(), trigger),
+    {ok, #state{check_intervals = CheckIntervals,
+                avs_path_and_servers_list = AvsPathAndServersList}}.
+
+
+%% @doc gen_server callback - Module:handle_call(Request, From, State) -> Result
+handle_call(stop,_From, State) ->
+    {stop, shutdown, ok, State};
+handle_call(_,_From, State) ->
+    {reply, ok, State}.
+
+%% @doc Handling cast message
+%% <p>
+%% gen_server callback - Module:handle_cast(Request, State) -> Result.
+%% </p>
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+
+%% @doc Handling all non call/cast messages
+%% <p>
+%% gen_server callback - Module:handle_info(Info, State) -> Result.
+%% </p>
+handle_info(trigger, #state{check_intervals = CheckIntervals,
+                            avs_path_and_servers_list = AvsPathAndServersList} = State) ->
+    ok = check_disk_space(AvsPathAndServersList),
+    erlang:send_after(CheckIntervals, self(), trigger),
+    {noreply, State};
+handle_info(_Info, State) ->
+    {noreply, State}.
+
+%% @doc This function is called by a gen_server when it is about to
+%%      terminate. It should be the opposite of Module:init/1 and do any necessary
+%%      cleaning up. When it returns, the gen_server terminates with Reason.
+terminate(_Reason,_State) ->
+    ok.
+
+%% @doc Convert process state when code is changed
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+
+%%====================================================================
+%% Internal Functions
+%%====================================================================
+check_disk_space([]) ->
+    ok;
+check_disk_space([#avs_path_and_servers{path = AVSPath,
+                                        servers = ServerList} | Rest]) ->
+    case is_freespace_lt_avs(AVSPath) of
```

Comment:
As **is_freespace_lt_avs** expects the argument is an actual file path for AVS, this PR is not working because AVSPath is set to a AVS directory. need to have one-to-one mapping for AVSPath and Server.

Reply:
@mocchira Fixed it